File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        Map<Character,Integer> charMap = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            if (!(charMap.containsKey(s.charAt(i)))) {
                charMap.put(s.charAt(i),i);
            } else {
                charMap.put(s.charAt(i),-1);
            }
        }

        int leastIndex = 1000000;

        for (Map.Entry<Character,Integer> entry : charMap.entrySet()) {
            if (entry.getValue() != -1) {
               if (entry.getValue() < leastIndex) {
                    leastIndex = entry.getValue();
               } 
            }
        }
        if (leastIndex == 1000000) {
            return -1;
        }
        return leastIndex;
    }
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        uniq_char = {}
        for char in s :
            if char in uniq_char :
                uniq_char[char] = -1
            else :
                uniq_char[char] = s.index(char)
        
        for char in s :
            if uniq_char[char] != -1 :
                return uniq_char[char]
        return -1
